### PR TITLE
Fix occasional app logouts

### PR DESCRIPTION
**Pull request by aforge**
Apparently, Tinode is unable to handle simultaneous connection requests.
Take care of this.
Also, make sure we can actually fetch messages in background following arrival of push notifications.

### CODE REVIEW

File: TinodeSDK/Tinode.swift [hunk 1]
```diff
@@ -1025,6 +1070,7 @@ public class Tinode {
             if connection == nil {
                 do {
                     try connect()
+                    return true
```
Comment (or-else) [inline TinodeSDK/Tinode.swift]: The log string below 
```
Tinode.log.error("Error: connection cannot be nil in reconnectNow()")
```
seems to be inconsistent with the new code.

Comment (aforge) [reply TinodeSDK/Tinode.swift]: Removed.

File: TinodeSDK/Tinode.swift [hunk 2]
```diff
@@ -912,40 +912,59 @@ public class Tinode {
...
-                }, onFailure: nil)
-                if tinode.autoLogin && reconnecting {
+                })
+                if doLogin {
                     try future?.then(
                         onSuccess: { [weak self] msg in
                             if let t = self?.tinode, let cred = t.loginCredentials, !t.loginInProgress {
-                                _ = self?.tinode.login(
-                                    scheme: cred.scheme, secret: cred.secret, creds: nil)
+                                return try self?.tinode.login(
```
[the remark this answers is not in the document]
Comment (aforge) [reply TinodeSDK/Tinode.swift]: Right. Thanks for pointing out.

File: TinodeSDK/Tinode.swift [hunk 3]
```diff
@@ -912,40 +912,59 @@ public class Tinode {
     }
     public class TinodeConnectionListener : ConnectionListener {
         var tinode: Tinode
+        var completionPromises = [PromisedReply<ServerMessage>]()
```
[the remark this answers is not in the document]
Comment (aforge) [reply TinodeSDK/Tinode.swift]: Done.

**Pull request merged**